packages/react-dom/src/__tests__/ReactFresh-test.internal.js

```diff
@@ -895,6 +895,597 @@ describe('ReactFresh', () => {
     }
   });
 
+  it('can preserve state for lazy after resolution', async () => {
+    if (__DEV__) {
+      let promise;
+      let OuterV1 = render(() => {
+        function Hello() {
+          const [val, setVal] = React.useState(0);
+          return (
+            <p style={{color: 'blue'}} onClick={() => setVal(val + 1)}>
+              {val}
+            </p>
+          );
+        }
+        __register__(Hello, 'Hello');
+
+        const Outer = React.lazy(() => {
+          promise = new Promise(resolve => {
+            setTimeout(() => resolve({default: Hello}), 100);
+          });
+          return promise;
+        });
+        __register__(Outer, 'Outer');
+
+        function App() {
+          return (
+            <React.Suspense fallback={<p>Loading</p>}>
+              <Outer />
+            </React.Suspense>
+          );
+        }
+
+        return App;
+      });
+
+      expect(container.textContent).toBe('Loading');
```

Comment:
why not use act() for this?
(and a couple more further down)

Reply:
Use how?

Reply:
I thought act doesn't flush Suspense. I need recipes :P

Reply:
replace 
```jsx
jest.runAllTimers();
await promise;
Scheduler.flushAll();
```

with 
```jsx
await act(async () => {
  jest.runAllTimers()
})
```

in general, we shouldn't have to call `Scheduler.flushAll()` by hand